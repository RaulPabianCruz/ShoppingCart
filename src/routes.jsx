import Container from './components/Container/Container';
import Home from './components/Home/Home';
import Shop from './components/Shop/Shop';
import Cart from './components/Cart/Cart';
import ErrorPage from './ErrorPage';

const routes = [
  {
    path: '/',
    element: <Container />,
    errorElement: <ErrorPage />,
    children: [
      { index: true, element: <Home /> },
      {
        path: 'shop',
        element: <Shop />,
      },
      {
        path: 'cart',
        element: <Cart />,
      },
    ],
  },
];

export default routes;
