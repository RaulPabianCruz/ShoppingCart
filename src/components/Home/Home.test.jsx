import React from 'react';
import { render, screen } from '@testing-library/react';
import { createMemoryRouter, RouterProvider } from 'react-router-dom';
import routes from '../../routes';

describe('Home page content', () => {
  describe('snapshot', () => {
    it('matches snapshot', () => {
      const router = createMemoryRouter(routes, { initialEntries: ['/'] });
      const { container } = render(<RouterProvider router={router} />);

      expect(container).toMatchSnapshot();
    });
  });

  describe('Home Page link', () => {
    it("Renders 'Browse goodies' link", () => {
      const router = createMemoryRouter(routes, { initialEntries: ['/'] });
      render(<RouterProvider router={router} />);

      const BrowseLink = screen.getByRole('link', {
        name: 'Browse the Goodies',
      });

      expect(BrowseLink).toBeInTheDocument();
      expect(BrowseLink).toHaveAttribute('href', '/shop');
    });
  });
});
