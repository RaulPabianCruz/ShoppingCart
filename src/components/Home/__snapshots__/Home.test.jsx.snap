// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home page content > snapshot > matches snapshot 1`] = `
<div>
  <div>
    <header
      class="_header_523845"
    >
      <img
        alt="The e-Mart Logo"
        class="_logo_523845 _headerItem_523845"
        src="/src/assets/eMartLogo.webp"
      />
      <h1
        class="_titleName_523845 _headerItem_523845"
      >
        The e-Mart!
      </h1>
      <a
        class="_links_523845 _headerItem_523845"
        href="/"
      >
        Home
      </a>
      <a
        class="_links_523845 _headerItem_523845"
        href="/shop"
      >
        Shop
      </a>
      <p
        class="_itemCount_523845 _headerItem_523845"
        data-testid="num-of-items"
      >
        0
      </p>
      <a
        class="_links_523845 _headerItem_523845"
        data-testid="cart-link"
        href="/cart"
      >
        <img
          alt="Cart"
          class="_icon_523845 _headerItem_523845"
          src="/src/assets/cart.svg"
        />
      </a>
    </header>
    <main>
      <div>
        <h1>
          The Catalog With Drip No Cap (Ong frfr)
        </h1>
        <p>
          The e-Mart contains some of the most 'swag' pieces you could ever hope to find. But we are not just about the drip, our catalog also has a wide selection of the 'useless junk/gag gift' items those Temu shoppers seem to love. This isn't your regular dropshipping site, its worse.
        </p>
        <a
          href="/shop"
        >
          Browse the Goodies
        </a>
      </div>
      <img
        alt=""
        src="/src/assets/ShoppingImg.jpg"
      />
    </main>
  </div>
</div>
`;
